cmd/soroban-rpc/internal/ingest/service.go

```diff
@@ -36,7 +36,7 @@ type Config struct {
 
 func NewService(cfg Config) *Service {
 	ctx, done := context.WithCancel(context.Background())
-	o := Service{
+	service := Service{
```

Comment:
I'd suggest making this into a `&Service` here so that subsequent calls would avoid copying the data structure.